src/Symfony/Component/Console/Application.php

```diff
@@ -1169,4 +1169,27 @@ private function splitStringByWidth($string, $width)
 
         return $lines;
     }
+
+    /**
+     * Returns all namespaces of the command name.
+     *
+     * @param string $name The full name of the command
+     *
+     * @return array The namespaces of the command
+     */
+    private function extractAllNamespaces($name)
+    {
+        $parts = explode(':', $name, -1);
```

Comment:
ah indeed. Can you add a comment in the code explaining that the -1 is here to skip the command short name when exploding ?